src/library_egl.js

```diff
@@ -281,17 +281,24 @@ var LibraryEGL = {
       }
       contextAttribs += 8;
     }
-    if (glesContextVersion != 2) {
+    if (glesContextVersion != 2 && glesContextVersion != 3) {
 #if GL_ASSERTIONS
-      Module.printErr('When initializing GLES2/WebGL1 via EGL, one must pass EGL_CONTEXT_CLIENT_VERSION = 2 to GL context attributes! GLES version ' + glesContextVersion + ' is not supported!');
+      Module.printErr('When initializing GLES2/WebGL1 via EGL, one must pass EGL_CONTEXT_CLIENT_VERSION = 2 or 3 to GL context attributes! GLES version ' + glesContextVersion + ' is not supported!');
 #endif
       EGL.setErrorCode(0x3005 /* EGL_BAD_CONFIG */);
       return 0; /* EGL_NO_CONTEXT */
     }
 
     _glutInitDisplayMode(0xB2 /* GLUT_RGBA | GLUT_DOUBLE | GLUT_DEPTH | GLUT_MULTISAMPLE | GLUT_STENCIL */);
+
     EGL.windowID = _glutCreateWindow();
     if (EGL.windowID != 0) {
+      if (glesContextVersion == 3 && (typeof WebGL2RenderingContext === 'undefined'
```

Comment:
Hmm, this is not good. We really need to migrate to using the new HTML5 API for context initialization, i.e. call https://github.com/kripken/emscripten/blob/master/src/library_gl.js#L408 instead. Would you like to do that change here?